resources/msys2_package.rb

```diff
@@ -76,30 +84,26 @@ def msys2_init
       cookbook 'mingw'
     end
 
-    # During first run, msys initialized the /home/user directory with files
-    # from /etc/skel.  It also does a number of other first time setup steps.
-    # The shell must be restarted and cannot be reused.
-    msys2_exec('msys2 first time init', 'exit') unless ::File.exist?(home_dir)
+    # During first run, msys initializes some pacman system state and mirror
+    # state.  It also creates a home directory for your user based on what
+    # $HOME is using files from /etc/skel.  The home-directory creation step
+    # will automatically be performed if other users log in - so if you wish
+    # to globally modify user first time setup, edit /etc/skel or add
+    # "post-setup" steps to /etc/post-install/
+    # The first-time init shell must be restarted and cannot be reused.
+    msys2_exec('msys2 first time init', 'exit') unless ::File.exist?(pacman_key_dir)
 
     # Update pacman and msys base packages.
-    # TODO: Why is this file not going away - debug this later.
-    # unless ::File.exist?(::File.join(root, 'usr/bin/update-core'))
-    unless ::File.exist?(updated_marker_file)
-      msys2_exec('sync msys2 packages', 'pacman -Sy')
-      msys2_exec('upgrade msys2 core packages', 'pacman -S --noconfirm --needed bash pacman msys2-runtime msys2-runtime-devel')
-      msys2_exec('upgrade entire msys2 system', 'pacman -Suu --noconfirm')
-      # Sometimes this step seems to be necessary.  Unsure why..
-      # Not doing this results in only a few packages getting upgraded.
-      msys2_exec('upgrade entire msys2 system', 'pacman -Syu --noconfirm')
-      file win_friendly_path(updated_marker_file) do
-        action :touch
-      end
+    if ::File.exist?(::File.join(root, 'usr/bin/update-core'))
```

Comment:
`usr/bin/update-core` doesn't get removed so this block executes every time and re-installs a bunch of stuff